internal/api/v1beta1/policy.go

```diff
@@ -107,18 +107,47 @@ func (h Handler) CreatePolicy(ctx context.Context, request *frontierv1beta1.Crea
 		return nil, grpcInternalServerError
 	}
 
-	audit.GetAuditor(ctx, schema.PlatformOrgID.String()).
-		LogWithAttrs(audit.PolicyCreatedEvent, audit.Target{
-			ID:   newPolicy.ResourceID,
-			Type: newPolicy.ResourceType,
-		}, map[string]string{
-			"role_id":        newPolicy.RoleID,
-			"principal_id":   newPolicy.PrincipalID,
-			"principal_type": newPolicy.PrincipalType,
-		})
+	auditPolicyCreationEvent(ctx, newPolicy)
 	return &frontierv1beta1.CreatePolicyResponse{Policy: policyPB}, nil
 }
 
+func (h Handler) CreatePolicyForProject(ctx context.Context, request *frontierv1beta1.CreatePolicyForProjectRequest) (*frontierv1beta1.CreatePolicyForProjectResponse, error) {
+	logger := grpczap.Extract(ctx)
+
+	if request.GetBody() == nil || request.GetBody().GetRoleId() == "" || request.GetBody().GetPrincipal() == "" {
+		return nil, grpcBadBodyError
+	}
+
+	principalType, principalID, err := schema.SplitNamespaceAndResourceID(request.GetBody().GetPrincipal())
+	if err != nil {
+		return nil, ErrNamespaceSplitNotation
+	}
+
+	p := policy.Policy{
+		RoleID:        request.GetBody().GetRoleId(),
+		PrincipalType: principalType,
+		PrincipalID:   principalID,
+		ResourceID:    request.GetProjectId(),
+		ResourceType:  schema.ProjectNamespace,
+	}
+
+	newPolicy, err := h.policyService.Create(ctx, p)
```

Comment:
Can you verify if this method converts project name to project id automatically? If not we should pass project id here if in case someone has passed project name?

Reply:
Updated